plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/registry/AIAssistantRegistry.java

```diff
@@ -47,22 +50,16 @@ public AIAssistantRegistry(IExtensionRegistry registry) {
         this.customDescriptor = customAssistantDescriptor;
     }
 
-    public <T extends AIAssistant> T getAssistant() {
-        if (globalAssistant == null) {
-            synchronized (this) {
-                if (globalAssistant == null) {
-                    if (customDescriptor != null) {
-                        try {
-                            globalAssistant = customDescriptor.createInstance();
-                        } catch (DBException e) {
-                            throw new IllegalStateException(e);
-                        }
-                    } else {
-                        globalAssistant = new AIAssistantImpl();
-                    }
-                }
-            }
+    @NotNull
+    public <T extends AIAssistant> T createAssistant(@NotNull DBPWorkspace workspace) throws DBException {
```

Comment:
So it won't cache an instance of the assistance per workspace, is this expected?

Reply:
Yes, it is a very thin wrapper